### PR TITLE
CcdCallbacks: manage timer-based cond callbacks properly

### PR DESCRIPTION
**Pull request by stwhite91**
Fix support for timer-based conditional callbacks that was broken by PR #3537,
while still minimizing timer calls when possible.

Also add a test for timer-based conditional callbacks.

### CODE REVIEW

**Review by evan-charmworks — APPROVED** [review]

File: src/conv-core/converse.h [hunk 1]
```diff
@@ -1785,34 +1792,36 @@ typedef void (*CcdCondFn)(void *userParam);
 typedef void (*CcdVoidFn)(void *userParam,double curWallTime);
 
 /*CPU conditions*/
-#define CcdPROCESSOR_BEGIN_BUSY 0
-#define CcdPROCESSOR_END_IDLE 0 /*Synonym*/
-#define CcdPROCESSOR_BEGIN_IDLE 1
-#define CcdPROCESSOR_END_BUSY 1 /*Synonym*/
-#define CcdPROCESSOR_STILL_IDLE 2
-#define CcdPROCESSOR_LONG_IDLE 3
+#define CcdSCHEDLOOP            0
```
Comment (ericjbohm) [inline src/conv-core/converse.h:1795]: Ideally this whitespace reformatting would be in a different commit, but it's unlikely to create problems here.

**Review by ericjbohm — APPROVED** [review]
